common/constants/default.go

```diff
@@ -147,3 +147,21 @@ const (
 	// MessageSuccessfulContent is the successful content value of Message struct
 	MessageSuccessfulContent string = "OK"
 )
+
+var (
+	SystemNamespace = SystemName
+	NamespaceEnvKey = "NAMESPACE"
+
+	once sync.Once
+)
+
+func init() {
+	once.Do(
+		func() {
+			ns := os.Getenv(NamespaceEnvKey)
+			if ns != "" {
+				SystemNamespace = ns
+			}
```

Comment:
How about parse from config file before running?

Reply:
Just have a see:
```
/ # cat /etc/kubeedge/config/cloudcore.yaml 
apiVersion: cloudcore.config.kubeedge.io/v1alpha2
kind: CloudCore
kubeAPIConfig:
  kubeConfig: ""
  master: ""
modules:
  cloudHub:
    advertiseAddress:
    - 192.168.88.6
    nodeLimit: 1000
    tlsCAFile: /etc/kubeedge/ca/rootCA.crt
    tlsCertFile: /etc/kubeedge/certs/edge.crt
    tlsPrivateKeyFile: /etc/kubeedge/certs/edge.key
    unixsocket:
      address: unix:///var/lib/kubeedge/kubeedge.sock
      enable: true
    websocket:
      address: 0.0.0.0
      enable: true
      port: 10000
    quic:
      address: 0.0.0.0
      enable: false
      maxIncomingStreams: 10000
      port: 10001
    https:
      address: 0.0.0.0
      enable: true
      port: 10002
  cloudStream:
    enable: true
    streamPort: 10003
    tunnelPort: 10004
  dynamicController:
    enable: false
  router:
    enable: false
```

But not have a namespace parameter.

Reply:
add one?

Reply:
Like this?
```
metadata:
   namespace: kubeedge
```
What do you think ? @fisherxu @gy95

Reply:
Add a config field in cloudcore for namespace maybe a little weird, as we already have the ns value from the pod. Can we add a new value in helm Value?

Reply:
Helm no need to add a namespace.  It already can work now. :)

Reply:
Okey, just seen helm have the --namespace option :) If users don't specify the --namespace option, helm will deploy the chart to `default` ns？

Reply:
> Okey, just seen helm have the --namespace option :) If users don't specify the --namespace option, helm will deploy the chart to `default` ns？

Yeah.

Reply:
We should provide a way for users of process deployment to customize the namespace, configuration files are obviously preferred.
@fisherxu @zhu733756

Reply:
Process deployment should be deprecated in future :) now most users all use deployment to deploy the cloudcore, how do you think? @zc2638  @gy95 @zhu733756

Reply:
Process deployment may use export NAMESPACE=kubeedge001. 
Read from cloudcore.yaml,  Cloudcore configmap need to be changed to
```
metadata:
   namespace: {{.Release.Namespace}}
```

Reply:
I think both are ok.